From c1e354ebad4f53389ade9fd8b474790887129d1b Mon Sep 17 00:00:00 2001
From: Guy Shimko <82650790+guyush1@users.noreply.github.com>
Date: Sat, 27 Dec 2025 07:41:03 +0200
Subject: [PATCH] Fix TIOCGWINSZ struct size mismatch (#1205)

A buffer the size of `struct winsize` should be passed to the TIOCGWINSZ
ioctl. The size of the winsize struct is 8 bytes (see
https://man7.org/linux/man-pages/man2/TIOCSWINSZ.2const.html), but a
4-bytes buffer was passed, which caused a BOF.

## Description

The patch fixes a BOF which caused a crash in some newer python
environments.

Fixes #1201
---
 gef.py | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/gef.py b/gef.py
index 7ce988f7a..08d193ab5 100644
--- a/gef.py
+++ b/gef.py
@@ -3796,7 +3796,7 @@ def get_terminal_size() -> tuple[int, int]:
         import fcntl
         import termios
         try:
-            tty_rows, tty_columns = struct.unpack("hh", fcntl.ioctl(1, termios.TIOCGWINSZ, "1234")) # type: ignore
+            tty_rows, tty_columns, _, _ = struct.unpack("hhhh", fcntl.ioctl(1, termios.TIOCGWINSZ, "12345678")) # type: ignore
             return tty_rows, tty_columns
         except OSError:
             return 600, 100
